File: app/src/test/java/de/bornholdtlee/defaultproject/ObjectBoxTest.java
package de.bornholdtlee.defaultproject;


import org.junit.Test;

import java.util.List;

import javax.inject.Inject;

import de.bornholdtlee.defaultproject.base.injection.BaseInjectTest;
import de.bornholdtlee.defaultproject.base.injection.TestComponent;
import de.bornholdtlee.defaultproject.model.DefaultModel;
import io.objectbox.Box;

import static junit.framework.Assert.assertEquals;

public class ObjectBoxTest extends BaseInjectTest {

    @Inject
    Box<DefaultModel> defaultModelBox;

    @Override
    public void inject(TestComponent testComponent) {
        testComponent.inject(this);
    }

    @Test
    public void shouldInsertModel() {
        List<DefaultModel> all = defaultModelBox.getAll();
        assertEquals(0, all.size());

        DefaultModel model = new DefaultModel();
        model.setName("hallo");

        defaultModelBox.put(model);

        all = defaultModelBox.getAll();
        assertEquals(1, all.size());
        assertEquals("hallo", all.get(0).getName());
    }
}

File: app/src/test/java/de/bornholdtlee/defaultproject/ObjectBoxTest.kt
package de.bornholdtlee.defaultproject


import de.bornholdtlee.defaultproject.base.BaseTest
import de.bornholdtlee.defaultproject.base.ITestInjection
import de.bornholdtlee.defaultproject.base.TestComponent
import de.bornholdtlee.defaultproject.model.DefaultModel
import io.objectbox.Box
import junit.framework.Assert.assertEquals
import org.junit.Test
import javax.inject.Inject

class ObjectBoxTest : BaseTest(), ITestInjection {

    @Inject
    lateinit var defaultModelBox: Box<DefaultModel>

    override fun inject(testComponent: TestComponent) {
        testComponent.inject(this)
    }

    @Test
    fun shouldInsertModel() {
        var all = defaultModelBox.all
        assertEquals(0, all.size)

        val model = DefaultModel()
        model.name = "hallo"

        defaultModelBox.put(model)

        all = defaultModelBox.all
        assertEquals(1, all.size)
        assertEquals("hallo", all[0].name)
    }
}
